core/trino-spi/src/main/java/io/trino/spi/ptf/ScalarArgument.java

```diff
@@ -38,7 +38,7 @@
     private final Object value;
 
     @JsonCreator
-    public ScalarArgument(@JsonProperty("type") Type type, @JsonProperty("value") Object value)
+    private ScalarArgument(@JsonProperty("type") Type type, @JsonProperty("value") Object value)
```

Comment:
Does `@JsonCreator` allow `private` constructor?

(we have some "jackson-only" public `@JsonCreator` methods, as if it wasn't allowed, but maybe we can close access to them)

Reply:
I tried to find out. All i found was that a private constructor might be chosen as a json creator without the annotation (e.g. preferred over a factory method).

Reply:
```
    public void testScalarArgumentJson()
    {
        ObjectMapperProvider objectMapperProvider = new ObjectMapperProvider();
        objectMapperProvider.setJsonDeserializers(ImmutableMap.of(Type.class, new MetadataUtil.TestingTypeDeserializer()));
        JsonCodecFactory codecFactory = new JsonCodecFactory(objectMapperProvider);

        JsonCodec<ScalarArgument> codec = codecFactory.jsonCodec(ScalarArgument.class);
        ScalarArgument sa = ScalarArgument.scalarArgumentBuilder()
                .type(INTEGER)
                .value(10)
                .build();

        String saString = codec.toJson(sa);
        ScalarArgument sa2 = codec.fromJson(saString);
        assertEquals(sa, sa2);
    }
```

fails with
```
Caused by: com.fasterxml.jackson.databind.exc.ValueInstantiationException: Cannot construct instance of `io.trino.spi.ptf.ScalarArgument`, problem: class com.fasterxml.jackson.databind.introspect.AnnotatedConstructor cannot access a member of class io.trino.spi.ptf.ScalarArgument with modifiers "private"
 at [Source: (String)"{"@type":"scalar","type":"integer","value":10}"; line: 1, column: 46]

```

Reply:
Oh, sorry I didn't check that. I'll change to public.